source/dofs/dof_accessor_set.cc

```diff
@@ -64,8 +91,26 @@ DoFCellAccessor<dim, spacedim, lda>::set_dof_values_by_interpolation(
           // or that you specify the correct one
           (fe_index == this->active_fe_index()) ||
           (fe_index == DoFHandler<dim, spacedim>::invalid_fe_index))
-        // simply set the values on this cell
-        this->set_dof_values(local_values, values);
+        {
+#ifdef DEBUG
+          Vector<number> tmp(this->get_fe().n_dofs_per_cell());
+          this->get_dof_values(values, tmp);
+
+          for (unsigned int i = 0; i < this->get_fe().n_dofs_per_cell(); ++i)
+            {
+              Assert(tmp[i] == number() ||
+                       get_abs(tmp[i] - local_values[i]) <=
+                         get_abs(tmp[i] + local_values[i]) * 100000. *
+                           std::numeric_limits<typename numbers::NumberTraits<
+                             number>::real_type>::epsilon(),
+                     ExcNonMatchingElementsSetDofValuesByInterpolation(
+                       local_values[i], tmp[i]));
```

Comment:
I think this change makes sense, as it is for a very specific context. Did you run the full test suite to see if any existing test triggers on this assertion?

Reply:
> Did you run the full test suite to see if any existing test triggers on this assertion?

Not yet. I first wanted to know if this is an acceptable change of behavior?

Reply:
I think the main question is how often we enter this function with a vector whose entries are not zero and should be overwritten (therefore my request). If there is a large number of failures, we might need to reconsider things and merely add a more descriptive error message to the parallel vector. That would have been the case for the usual `set_dof_values` function where I think the break would be overwhelming. But for the present context I believe things could be more specific and hence worthwhile. But let's wait for other opinions as well.

Reply:
Reading through this, I don't really see why one would *need* to start with a zero vector. It makes good sense to me to start with whatever vector one just has sitting around, and this may also be used to zero out components of a vector, or set them to different values. Think about fixing up the pressure so that its mean value is zero by getting the values, modifying the pressure, and setting them again. In essence, it seems gratuitous to make a difference between the current function and `set_dof_values()` -- the two of them should be the same on active cells (and are, currently), and this patch makes them different.